### ISSUE TITLE
ftw crashes when alpha-blending is exported

### ISSUE THREAD

**Issue by dertuxmalwieder** [fjames86/ftw#6]
A curious little problem: I wrote an application with `ftw` and it *generally* works just fine (thank you for your efforts!). However, creating and using an .exe just won't work *if* I use alpha-blending. I can perfectly reproduce that with your own example code, as `icon.lisp` uses alpha-blending as well:

```
> sbcl --load icon.lisp

(...)

* (sb-ext:save-lisp-and-die "test.exe" :toplevel #'ftw.icon:test-icon3 :executable t)
```

Trying to run test.exe, the result I get in the console window is: "Error 87: wrong parameter". This is weird, as calling `(ftw.icon:test-icon3)` right from inside of the REPL does not have any problem, and I have no problem compiling other examples like `dragdrop.lisp`. What could be the culprit here?

Windows 10, 64-bit.

**Comment by fjames86** [comment]
Good question, and off the top of my head I've no idea. Quite an odd bug. I admit I never used saved images so it's not something I ever tested when I wrote it. 

I will try to reproduce it.

**Comment by dertuxmalwieder** [comment]
Ah, I could perfectly live without the alpha blending (and/or .exe files), but the tool I wrote *could* be useful for other people, and those other people like 1) eye-candy and 2) pre-built binaries...

Thank you for looking into this!

**Comment by fjames86** [comment]
Ok I finally got some time to look at this. I reproduced it and identified the issue. It seems if you allocate a handle then save-lisp-and-die then all your handles are invalid. You need to instead save data arrays and allocate the handles in some sort of initialisation routine after saving the image. I updated the icon.lisp example so it now works in this scenario but it likely affects all such APIs.

**Comment by dertuxmalwieder** [comment]
Thank you a lot! (I know that FLOSS developers rarely get enough credits, so here's mine.)

What would be the generic way to "fix" the applications now? I looked at the diff and it looks like it's rather specific.

**Comment by fjames86** [comment]
I think you need to create all your bitmaps/icons etc. after restarting from save-lisp-and-die. Perhaps change from allocating directly in defvar/defparameter to some sort of lazy initialisation where it is allocated only on first use.

**Comment by dertuxmalwieder** [comment]
I'll try that next week. Thank you!

**Comment by dertuxmalwieder** [comment]
Indeed, using the bitmap/icon data as `defvar`s and calling them from a `defun` during `+wm-paint+` works. :-)
